### PR TITLE
[backport][SES5] qa/deploy: use jq instead of python in salt_api_test

### PR DESCRIPTION
**Pull request by smithfarm**
Backport of https://github.com/SUSE/DeepSea/pull/1377

### CODE REVIEW

File: qa/common/common.sh [hunk 1]
```diff
@@ -267,7 +267,7 @@ function ceph_health_test {
 function salt_api_test {
     echo "Salt API test: BEGIN"
     systemctl --no-pager --full status salt-api.service
-    curl http://$(hostname):8000/ | python3 -m json.tool
+    curl http://$(hostname):8000/ | python -m json.tool
```
Comment (kshtsk) [inline qa/common/common.sh]: Is this the same as `| jq`?

Comment (smithfarm) [reply qa/common/common.sh]: Yes, that would work fine as well.

**Comment by smithfarm** [comment]
@kshtsk Please take another look.

**Comment by smithfarm** [comment]
Note that `suse/tier2/openattic` is not reproducing the problem that is seen in the openATTIC testing Jenkins job, because (presumably) the teuthology environment for testing SES5 has python3 installed in it, whereas the openATTIC testing Jenkins environment (presumably) does not.

**Comment by kshtsk** [comment]
@susebot run teuthology for deepsea/tier2/openattic

**Comment by kshtsk** [comment]
@smithfarm yes it is not reproducing the issue, but hypothetically it can be used to test the PR, because 'python3' replaced with 'jq'.

**Comment by smithfarm** [comment]
@kshtsk `suse/tier1` is sufficient, since the `salt_api_test` function is run in all the tests

**Comment by smithfarm** [comment]
@susebot run teuthology

**Comment by smithfarm** [comment]
Works:

```
2018-09-08T06:59:43.932 INFO:teuthology.orchestra.run.target137074024006.stderr:+ salt_api_test
2018-09-08T06:59:43.933 INFO:teuthology.orchestra.run.target137074024006.stderr:++ mktemp
2018-09-08T06:59:43.933 INFO:teuthology.orchestra.run.target137074024006.stderr:+ local tmpfile=/tmp/tmp.GXeLwl8yx5
2018-09-08T06:59:43.933 INFO:teuthology.orchestra.run.target137074024006.stderr:+ echo 'Salt API test: BEGIN'
2018-09-08T06:59:43.933 INFO:teuthology.orchestra.run.target137074024006.stderr:+ systemctl --no-pager --full status salt-api.service
2018-09-08T06:59:43.933 INFO:teuthology.orchestra.run.target137074024006.stdout:Salt API test: BEGIN
2018-09-08T06:59:43.938 INFO:teuthology.orchestra.run.target137074024006.stdout:â— salt-api.service - The Salt API
2018-09-08T06:59:43.939 INFO:teuthology.orchestra.run.target137074024006.stdout:   Loaded: loaded (/usr/lib/systemd/system/salt-api.service; enabled; vendor preset: disabled)
2018-09-08T06:59:43.939 INFO:teuthology.orchestra.run.target137074024006.stdout:   Active: active (running) since Sat 2018-09-08 06:56:25 UTC; 3min 18s ago
2018-09-08T06:59:43.939 INFO:teuthology.orchestra.run.target137074024006.stdout: Main PID: 9790 (salt-api)
2018-09-08T06:59:43.939 INFO:teuthology.orchestra.run.target137074024006.stdout:    Tasks: 104 (limit: 512)
2018-09-08T06:59:43.940 INFO:teuthology.orchestra.run.target137074024006.stdout:   CGroup: /system.slice/salt-api.service
2018-09-08T06:59:43.940 INFO:teuthology.orchestra.run.target137074024006.stdout:           â”œâ”€9790 /usr/bin/python /usr/bin/salt-api
2018-09-08T06:59:43.940 INFO:teuthology.orchestra.run.target137074024006.stdout:           â””â”€9809 /usr/bin/python /usr/bin/salt-api
2018-09-08T06:59:43.940 INFO:teuthology.orchestra.run.target137074024006.stdout:
2018-09-08T06:59:43.940 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [DEBUG   ] Process Manager starting!
2018-09-08T06:59:43.940 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
2018-09-08T06:59:43.940 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
2018-09-08T06:59:43.941 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [INFO    ] [08/Sep/2018:06:56:25] ENGINE Listening for SIGHUP.
2018-09-08T06:59:43.941 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [INFO    ] [08/Sep/2018:06:56:25] ENGINE Listening for SIGTERM.
2018-09-08T06:59:43.941 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [INFO    ] [08/Sep/2018:06:56:25] ENGINE Listening for SIGUSR1.
2018-09-08T06:59:43.941 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [INFO    ] [08/Sep/2018:06:56:25] ENGINE Bus STARTING
2018-09-08T06:59:43.941 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [INFO    ] [08/Sep/2018:06:56:25] ENGINE Started monitor thread '_TimeoutMonitor'.
2018-09-08T06:59:43.941 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [INFO    ] [08/Sep/2018:06:56:25] ENGINE Serving on http://0.0.0.0:8000
2018-09-08T06:59:43.941 INFO:teuthology.orchestra.run.target137074024006.stdout:Sep 08 06:56:25 target137074024006.teuthology salt-api[9790]: [INFO    ] [08/Sep/2018:06:56:25] ENGINE Bus STARTED
2018-09-08T06:59:43.942 INFO:teuthology.orchestra.run.target137074024006.stderr:+ tee /tmp/tmp.GXeLwl8yx5
2018-09-08T06:59:43.942 INFO:teuthology.orchestra.run.target137074024006.stderr:++ hostname
2018-09-08T06:59:43.942 INFO:teuthology.orchestra.run.target137074024006.stderr:+ curl http://target137074024006.teuthology:8000/
2018-09-08T06:59:43.950 INFO:teuthology.orchestra.run.target137074024006.stderr:  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2018-09-08T06:59:43.950 INFO:teuthology.orchestra.run.target137074024006.stderr:                                 Dload  Upload   Total   Spent    Left  Speed
2018-09-08T06:59:43.963 INFO:teuthology.orchestra.run.target137074024006.stderr:
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   146  100   146    0     0  10438      0 --:--:-- --:--:-- --:--:-- 11230
2018-09-08T06:59:43.964 INFO:teuthology.orchestra.run.target137074024006.stderr:+ test -s /tmp/tmp.GXeLwl8yx5
2018-09-08T06:59:43.964 INFO:teuthology.orchestra.run.target137074024006.stderr:+ jq . /tmp/tmp.GXeLwl8yx5
2018-09-08T06:59:43.970 INFO:teuthology.orchestra.run.target137074024006.stderr:+ echo -e '\n'
2018-09-08T06:59:43.970 INFO:teuthology.orchestra.run.target137074024006.stderr:+ rm /tmp/tmp.GXeLwl8yx5
2018-09-08T06:59:43.971 INFO:teuthology.orchestra.run.target137074024006.stdout:{"clients": ["local", "local_async", "local_batch", "local_subset", "runner", "runner_async", "ssh", "wheel", "wheel_async"], "return": "Welcome"}
2018-09-08T06:59:43.971 INFO:teuthology.orchestra.run.target137074024006.stdout:
2018-09-08T06:59:43.973 INFO:teuthology.orchestra.run.target137074024006.stdout:
2018-09-08T06:59:43.973 INFO:teuthology.orchestra.run.target137074024006.stdout:Salt API test: END
```

**Comment by smithfarm** [comment]
log output now looks like this:

```
2018-09-08T07:26:21.746 INFO:teuthology.orchestra.run.target137074025158.stderr:+ tee /tmp/tmp.rSwprfSTW1
2018-09-08T07:26:21.746 INFO:teuthology.orchestra.run.target137074025158.stderr:++ hostname
2018-09-08T07:26:21.746 INFO:teuthology.orchestra.run.target137074025158.stderr:+ curl http://target137074025158.teuthology:8000/
2018-09-08T07:26:21.753 INFO:teuthology.orchestra.run.target137074025158.stderr:  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2018-09-08T07:26:21.753 INFO:teuthology.orchestra.run.target137074025158.stderr:                                 Dload  Upload   Total   Spent    Left  Speed
2018-09-08T07:26:21.762 INFO:teuthology.orchestra.run.target137074025158.stderr:
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   146  100   146    0     0  15275      0 --:--:-- --:--:-- --:--:-- 16222
2018-09-08T07:26:21.762 INFO:teuthology.orchestra.run.target137074025158.stderr:+ test -s /tmp/tmp.rSwprfSTW1
2018-09-08T07:26:21.763 INFO:teuthology.orchestra.run.target137074025158.stderr:+ jq . /tmp/tmp.rSwprfSTW1
2018-09-08T07:26:21.767 INFO:teuthology.orchestra.run.target137074025158.stdout:{"clients": ["local", "local_async", "local_batch", "local_subset", "runner", "runner_async", "ssh", "wheel", "wheel_async"], "return": "Welcome"}
2018-09-08T07:26:21.768 INFO:teuthology.orchestra.run.target137074025158.stderr:+ echo -en '\n'
2018-09-08T07:26:21.768 INFO:teuthology.orchestra.run.target137074025158.stderr:+ rm /tmp/tmp.rSwprfSTW1
2018-09-08T07:26:21.769 INFO:teuthology.orchestra.run.target137074025158.stdout:Salt API test: END
```

**Comment by smithfarm** [comment]
@susebot run teuthology

**Comment by smithfarm** [comment]
Touches `qa/` only - merging upon agreement in the DS call.

**Pull request merged**